lib/rucio/core/replica_sorter.py

```diff
@@ -228,7 +245,11 @@ def __get_distance_custom(rse: Union[tuple, str], client_location: dict) -> floa
     return cache_val
 
 
-def site_selector(replicas, site, vo):
+def site_selector(
+        replicas: dict[str, Any],
```

Comment:
Is there really a dedicated replica dictionary? I think that would make more sense

Reply:
Addressed in latest revision - created a `ReplicaDict` type